code/modules/mob/mob.dm

```diff
@@ -500,6 +500,41 @@ var/list/slot_equipment_priority = list( \
 		return 0 //Unsupported slot
 		//END HUMAN
 
+/mob/proc/get_visible_mobs()
+	var/list/seen_mobs = list()
+	for(var/mob/M in view(src))
+		seen_mobs += M
+
+	return seen_mobs
+
+/**
+ * Returns an assoc list which contains the mobs in range and their "visible" name.
+ * Mobs out of view but in range will be listed as unknown. Else they will have their visible name
+*/
+/mob/proc/get_telepathic_targets()
+	var/list/validtargets = new /list()
+	var/turf/T = get_turf(src)
+	var/list/mobs_in_view = get_visible_mobs()
+
+	for(var/mob/living/M in range(14, T))
```

Comment:
Since `range()` returns a list.. is it possible to do something like:
```dm
for(var/mob/living/M in range(14, T) - src)
```
and then skip the:
```dm
    if(M == src)
        continue
```
?

Reply:
Interesting idea. I've benchmarked both and the check version is slightly faster. about 69 MS vs 56 MS.
I'll keep it how it is now to be more conform to other checks as well

Reply:
Oh, sad..